haystack/document_stores/weaviate.py

```diff
@@ -495,7 +495,7 @@ def write_documents(
                     docs_batch.add(_doc, class_name=index, uuid=doc_id, vector=vector)
```

Comment:
If you plan using `weaviate-client==3.3.3`, instead of `docs_batch.add` use `self.weaviate_client.batch.add_data_object`. So it should look like this:
```python
self.weaviate_client.batch.add_data_object(
    data_object=_doc,
    class_name=index,
    uuid=doc_id,
    vector=vector,
)
```